File: java/src/main/java/com/experian/dvs/client/address/format/enrichment/Geocodes.java
package com.experian.dvs.client.address.format.enrichment;

import com.experian.dvs.client.address.format.GeocodeMatchLevel;
import com.experian.dvs.client.server.address.format.RestApiEnrichmentDatasetGeocodes;

public class Geocodes {

    private final Double latitude;
    private final Double longitude;
    private final GeocodeMatchLevel matchLevel;

    public Geocodes(final RestApiEnrichmentDatasetGeocodes geocodes) {
        this.latitude = geocodes.getLatitude();
        this.longitude = geocodes.getLongitude();
        this.matchLevel = geocodes.getMatchLevel() != null ? GeocodeMatchLevel.fromValue(geocodes.getMatchLevel()) : null;
    }

    public GeocodeMatchLevel getMatchLevel() {
        return matchLevel;
    }

    public Double getLongitude() {
        return longitude;
    }

    public Double getLatitude() {
        return latitude;
    }
}

File: csharp/DVSClient/DVSClient/address/format/enrichment/Geocodes.cs
using DVSClient.Common;
using DVSClient.Server.Address.Format;

namespace DVSClient.Address.Format.Enrichment
{
    public class Geocodes
    {
        public double? Latitude { get; }
        public double? Longitude { get; }
        public GeocodeMatchLevel? MatchLevel { get; }

        public Geocodes(RestApiEnrichmentDatasetGeocodes geocodes)
        {
            Latitude = geocodes.Latitude;
            Longitude = geocodes.Longitude;
            MatchLevel = geocodes.MatchLevel?.GetEnumValueFromJsonName<GeocodeMatchLevel>();
        }
    }
}